add option to use pixi instead of docker

pablovela5620 opened the pull request:
should make it much easier to run the example

pablovela5620 committed:
add option to use pixi

02alexander commented:
Nice! It looks good, but the pixi environment doesn't quite work on my manjaro linux machine. 
When I run `pixi run example` it runs into error related to `rerun::Image`, it looks like it uses the rerun version from `FetchContent` by default (which is a too new rerun version) instead of `librerun-sdk` that's installed via pixi. And if I remove the following lines in `glomap/CMakeLists.txt` it works
```cmake
include(FetchContent)
FetchContent_Declare(rerun_sdk URL
    https://github.com/rerun-io/rerun/releases/latest/download/rerun_cpp_sdk.zip)
FetchContent_MakeAvailable(rerun_sdk)
```

So I think we should put the  `FetchContent` should be behind a flag with, like this:
```cmake
if (FETCH_RERUN)
  include(FetchContent)
  FetchContent_Declare(rerun_sdk URL
      https://github.com/rerun-io/rerun/releases/download/0.17.0/rerun_cpp_sdk.zip)
  FetchContent_MakeAvailable(rerun_sdk)
endif()
```
and add a flag in `/CMakeLists.txt`
```cmake
# ...
option(FETCH_COLMAP "Whether to use COLMAP with FetchContent or with self-installed software" ON)
option(FETCH_POSELIB "Whether to use PoseLib with FetchContent or with self-installed software" ON)
option(FETCH_RERUN "Whether to use Rerun with FetchContent or with self-installed software" ON)

```
And just disable it in the pixi environment in `pixi.toml`
```bash
cmake -B build -S . -GNinja \
    -DCMAKE_PREFIX_PATH=$CONDA_PREFIX \
    -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX \
    -DCMAKE_EXE_LINKER_FLAGS="-L$CONDA_PREFIX/lib" \
    -DFETCH_COLMAP=OFF \
    -DFETCH_POSELIB=OFF \
    -DFETCH_RERUN=OFF
```

However, on my NixOS machine it doesn't manage to run the example, even with the correct rerun version, I just get the following error when running:
```
CMake Error at .pixi/envs/default/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY OPENGL_glx_LIBRARY
  OPENGL_INCLUDE_DIR)
Call Stack (most recent call first):
  .pixi/envs/default/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  .pixi/envs/default/share/cmake-3.30/Modules/FindOpenGL.cmake:579 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  .pixi/envs/default/share/colmap/cmake/FindDependencies.cmake:32 (find_package)
  .pixi/envs/default/share/colmap/colmap-config.cmake:106 (include)
  cmake/FindDependencies.cmake:25 (find_package)
  CMakeLists.txt:18 (include)
```
But this might be just because NixOS is weird, we shouldn't worry to much about this, but I'm curious if you have any idea what might be the cause of this.

02alexander committed:
Add unzip and wget as dependencies

unzip and wget are used when fetching the data so they should be included as dependencies.

pablovela5620 committed:
update to allow using fetched rerun or rerun from pixi

pablovela5620 committed:
Merge branch 'main' of github.com:pablovela5620/glomap

pablovela5620 commented:
@02alexander as recommended I added a fetchcontent flag for rerun! I'm not totally sure what is going on with NixOS, but I've gotten it working on my ubuntu machines so far
